routes/api/v1/foods.js

```diff
@@ -44,4 +44,29 @@ router.get('/:id', async function(req, res, next){
   }
 })
 
+router.patch('/:id', async function(req, res, next){
+  try {
+    let food = await Food.findByPk(req.params.id)
+    let name = req.body.food.name
+    let calories = req.body.food.calories
+    if(name, calories == undefined ) {
+      reject()
+    } else {
+      let new_food = await food.update({
+        name: name,
+        calories: calories
+      });
+      if(new_food != null) {
+        res.setHeader(...defaultHeader);
+        res.status(200).send(JSON.stringify(new_food))
+      } else {
+        reject();
+      }
+    }
+  } catch {
+    res.setHeader(...defaultHeader);
+    res.status(400).send({ error: "Invalid ID or name/calories missing."})
```

Comment:
How / where does this go to the catch when an invalid ID is passed?

Reply:
Catch has been updated and error handling robustified.